src/assets/wise5/directives/teacher-summary-display/dialog-guidance-teacher-summary-display.component.ts

```diff
@@ -0,0 +1,119 @@
+import { AnnotationService } from '../../services/annotationService';
+import { CommonModule } from '@angular/common';
+import { Component } from '@angular/core';
+import { ComponentState } from '../../../../app/domain/componentState';
+import { ConfigService } from '../../services/configService';
+import { CRaterIdea } from '../../components/common/cRater/CRaterIdea';
+import { CRaterRubric } from '../../components/common/cRater/CRaterRubric';
+import { CRaterService } from '../../services/cRaterService';
+import { MatCardModule } from '@angular/material/card';
+import { MatIconModule } from '@angular/material/icon';
+import { SummaryService } from '../../components/summary/summaryService';
+import { TeacherDataService } from '../../services/teacherDataService';
+import { TeacherProjectService } from '../../services/teacherProjectService';
+import { TeacherSummaryDisplayComponent } from './teacher-summary-display.component';
+
+@Component({
+  imports: [CommonModule, MatCardModule, MatIconModule],
+  selector: 'dialog-guidance-teacher-summary-display',
+  styles: `
+    .wrapper {
+      display: grid;
+      grid-template-columns: 400px 400px;
+    }
+    .idea-group {
+      padding-left: 20px;
+    }
+    #all-ideas {
+      padding-top: 20px;
+    }
+  `,
+  templateUrl: 'dialog-guidance-teacher-summary-display.component.html'
+})
+export class DialogGuidanceTeacherSummaryDisplayComponent extends TeacherSummaryDisplayComponent {
+  protected ideaCountMap: Map<string, Set<number>> = new Map<string, Set<number>>();
+  protected seeAllIdeas: boolean;
+  protected allIdeas: { id: string; text: string; count: number }[] = [];
+  protected mostCommonIdeas: { id: string; text: string; count: number }[] = [];
+  protected leastCommonIdeas: { id: string; text: string; count: number }[] = [];
+  private rubric: CRaterRubric;
+
+  constructor(
+    protected annotationService: AnnotationService,
+    protected configService: ConfigService,
+    private cRaterService: CRaterService,
+    protected dataService: TeacherDataService,
+    protected projectService: TeacherProjectService,
+    protected summaryService: SummaryService
+  ) {
+    super(annotationService, configService, dataService, projectService, summaryService);
+  }
+
+  ngOnInit(): void {
+    this.rubric = this.cRaterService.getCRaterRubric(this.nodeId, this.componentId);
+    this.getLatestWork().subscribe((componentStates) => {
+      this.extractIdeas(componentStates);
+      const sortedIdeas = this.sortIdeas();
+      this.mostCommonIdeas = [...sortedIdeas].splice(0, 3);
+      this.leastCommonIdeas = [...sortedIdeas]
+        .splice(sortedIdeas.length - 3, sortedIdeas.length)
+        .reverse();
+      this.allIdeas = this.getAllIdeas();
+      if (!this.allIdeas.some((idea) => this.ideaCountMap.get(idea.id)?.size > 0)) {
+        this.doRender = false;
+      }
```

Comment:
Don't sort ideas and find most/least common ideas if we don't have ideas?
```suggestion
      if (!this.allIdeas.some((idea) => this.ideaCountMap.get(idea.id)?.size > 0)) {
        this.doRender = false;
      } else {
	this.allIdeas = this.getAllIdeas();
        const sortedIdeas = this.sortIdeas();
        this.mostCommonIdeas = [...sortedIdeas].splice(0, 3);
        this.leastCommonIdeas = [...sortedIdeas]
          .splice(sortedIdeas.length - 3, sortedIdeas.length)
          .reverse();
      }
```